### ISSUE TITLE
Image not uploaded when copying full document content

### ISSUE THREAD

**Issue by fubar-coder** [ckeditor/ckeditor5-paste-from-office#51]
The image is not uploaded when copying the full contents of the Word document.

When copying only the image with the surrounding paragraphs, the image is uploaded again.

Test Word file:
[PasteFromWordTest.docx](https://github.com/ckeditor/ckeditor5-paste-from-office/files/2844717/PasteFromWordTest.docx)

**Comment by Reinmar** [comment]
cc @Mgsy @f1ames

**Comment by Mgsy** [comment]
Hi, @fubar-coder, I've just tested PFO with the provided document and I'm not able to reproduce this issue. The image was uploaded properly (using Word for Office 365, version 1901).  

Could you give us more information about your development environment or show us your CKEditor 5 config?

**Comment by fubar-coder** [comment]
I used the CKEditor 5 demo (classic): https://ckeditor.com/ckeditor-5/demo/

Other than that, I cannot tell you much more except that I used Word 1808.

**Comment by Mgsy** [comment]
Unfortunately, I'm still not able to reproduce your issue - the content is pasted properly in all demos (checked on Windows/Mac in all browsers). 

Could you record a screencast which shows exact steps to reproduce? Additionally, you can check the console whether there is any error.

**Comment by fubar-coder** [comment]
FYI: I used Chrome 71.0.3578.98.

**Comment by Mgsy** [comment]
@f1ames, is it possible that the Word version (Word 2016, version 1808) could generate some problems with pasting images?

@fubar-coder, could you check whether you can reproduce it in our [Paste from Word sample](https://ckeditor.com/docs/ckeditor5/latest/features/paste-from-word.html)? Also, please open the console to spot any errors.

**Comment by fubar-coder** [comment]
The problem is reproducible. I attached the console log.
[ckeditor.com-1549966746009.log](https://github.com/ckeditor/ckeditor5-paste-from-office/files/2855252/ckeditor.com-1549966746009.log)

**Comment by f1ames** [comment]
Works for me too - Word 16.21 (190117) on Chrome 71.0.3578.98 (macOS).

When tested with [Paste from Word sample](https://ckeditor.com/docs/ckeditor5/latest/features/paste-from-word.html) I got `Error: Illegal file mime type` from `Easy Image` plugin (upload error, seems like `.gif` is not supported), but the image itself is inserted correctly:

![image](https://user-images.githubusercontent.com/1061942/52633856-eeb84700-2ec5-11e9-8ce9-6931aac353c5.png)

so this is something different.

Looking at the log attached by @fubar-coder it looks like the image is not correctly transformed into `base64` representation. I'm not sure why it does not happen in our environments 🤔 

---

To clarify, @fubar-coder you were able to reproduce the issue on [Paste from Word sample](https://ckeditor.com/docs/ckeditor5/latest/features/paste-from-word.html) with Chrome? What OS do you use?

I would like to ask you to try with [this codepen sample](https://codepen.io/f1ames/pen/omyzQr) to see if it works. And then, after pasting, if you could share the content of `Clipboard HTML` container (below editor) e.g. as [gist](https://gist.github.com/) (because it will be quite large).

**Comment by fubar-coder** [comment]
Yes, I can reproduce this problem with the given sample. I use Windows 10, 1809, Build 17763.253.

This is the gist which contains two files: one for Chrome (with the error) and one for Firefox (succeeding)

https://gist.github.com/fubar-coder/10a04caa0fa4d6fd4da48076f5fb5740

**Comment by f1ames** [comment]
Thanks for clarification and sharing output code @fubar-coder!

The strange thing is that the output is basically identical for both browsers (apart from the empty line on the beginning which is always added in Firefox) - see the diff https://www.diffchecker.com/u9nwqdkS.

At least we know that the issues is not Word clipboard output itself. I assume the case might be that for some reason extracting image RTF data fails in Chrome. One cause may be the fact that extraction function checks for `jpg/png` like types in RTF data:

https://github.com/ckeditor/ckeditor5-paste-from-office/blob/cf9353f43792e5197ce6564a9475301556523036/src/filters/image.js#L187-L191

Maybe with particular Windows/Word versions the type is different 🤔 

Since we are not able to reproduce it on Windows with the similar setup, I would ask you @fubar-coder for one more thing. Same as before if you could try with [this codepen sample](https://codepen.io/f1ames/pen/omMZJm) and share whole `Images RTF` data as a gist, I would greatly appreciate.

**Comment by fubar-coder** [comment]
Chrome didn't show anything in the "Images RTF". I also created a Gist for "Images RTF" while using Firefox:

https://gist.github.com/fubar-coder/7a66caac06db86ed0d5fc7f71ba727bc

**Comment by fubar-coder** [comment]
Some additional note: data.dataTransfer.getData( 'text/rtf' ) returns an empty string 😕

**Comment by f1ames** [comment]
Thanks again @fubar-coder.

> Some additional note: `data.dataTransfer.getData( 'text/rtf' )` returns an empty string

That's strange TBH and since you are using recent version of the Chrome browser https://bugs.chromium.org/p/chromium/issues/detail?id=317807 should be fixed. Is it possible that you have any active 3rd-party software or a Chrome extensions which may manipulate clipboard contents (you may try to run the sample with all extensions turned off if you have any to check it)?

**Comment by fubar-coder** [comment]
Yes, I tried both the "Incognito Window" and also disabled all extensions and restarted the browser, but the problem stays.

**Comment by Mgsy** [comment]
Here is another case reported - https://github.com/ckeditor/ckeditor5/issues/1710.

Try to download this document and paste it to the editor in Chrome. Images won't paste.

[123.docx](https://github.com/ckeditor/ckeditor5-paste-from-office/files/3116170/123.docx)

**Comment by Mgsy** [comment]
The same problem with an identical error message - https://github.com/ckeditor/ckeditor5/issues/1710 (I can't reproduce it).

Here is the file from the ticket:
[123.docx](https://github.com/ckeditor/ckeditor5-paste-from-office/files/3126294/123.docx)

**Comment by Hubmayun123** [comment]
> 是的，我尝试了“隐身窗口”并禁用了所有扩展并重新启动了浏览器，但问题仍然存在。

I had the same problem.How did you finally solve it ?

**Comment by msamsel** [comment]
I see that I can reproduce such issue, when image is located inside a table. In such scenario I couldn't get an image pasted in CKEditor.

**Comment by Mgsy** [comment]
Another case reported - https://github.com/ckeditor/ckeditor5-paste-from-office/issues/60.

**Comment by lasfin** [comment]
Doesn't work for me too. Maybe it's because MS Word version is 2013.

**Comment by Mgsy** [comment]
@lasfin, can I ask you to provide a sample file which allows you to reproduce the problem? Also, it will be great if you can copy the content from this file, paste it to this [codepen sample](https://codepen.io/f1ames/pen/omyzQr) and share result from `Clipboard HTML`.

**Comment by lasfin** [comment]
@Mgsy 
[test.docx](https://github.com/ckeditor/ckeditor5-paste-from-office/files/3447389/test.docx)

[clipboard.txt](https://github.com/ckeditor/ckeditor5-paste-from-office/files/3447414/clipboard.txt)
